adriatic crashed with 2.1.5 candidate

```
W190214 14:39:14.922943 225998271 util/log/clog.go:1400  log traffic is now intercepted; log files will be incomplete
panic: runtime error: index out of range

goroutine 226007656 [running]:
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/trace.(*trace).addEvent(0xc43edd91e0, 0x2735280, 0xc42b4bfb90, 0xc442220000)
        /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/trace/trace.go:772 +0x734
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/trace.(*trace).LazyPrintf(0xc43edd91e0, 0x2ad856f, 0x2, 0xc44222b8c0, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/trace/trace.go:801 +0xa0
github.com/cockroachdb/cockroach/pkg/util/tracing.(*span).LogFields(0xc44509cb60, 0xc43fab9c80, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/tracer_span.go:389 +0x65e
github.com/cockroachdb/cockroach/pkg/util/log.eventInternal(0x305a6a0, 0xc42b4bf890, 0x2680000, 0x2ae0906, 0x8, 0xc446599b10, 0x3, 0x3)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:132 +0x21c
github.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x305a6a0, 0xc42b4bf890, 0x1, 0x2, 0x2b1f9a4, 0x22, 0xc446599c40, 0x2, 0x2)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:84 +0x298
github.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x305a6a0, 0xc42b4bf890, 0x1, 0xc400000001, 0x2b1f9a4, 0x22, 0xc446599c40, 0x2, 0x2)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:69 +0x8c
github.com/cockroachdb/cockroach/pkg/util/log.Infof(0x305a6a0, 0xc42b4bf890, 0x2b1f9a4, 0x22, 0xc446599c40, 0x2, 0x2)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:87 +0x7e
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).releaseBudget(0xc447768be0, 0x305a6a0, 0xc42b4bf890)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:680 +0x134
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).doStop(0xc447768be0, 0x305a6a0, 0xc42b4bf890, 0xc42b4bf801)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:405 +0x94
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).Stop(0xc447768be0, 0x305a6a0, 0xc42b4bf890)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:384 +0x44
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*sorterBase).close(0xc43921e000)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/sorter.go:145 +0xfc
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.newSortAllProcessor.func1(0x305a6a0, 0xc42b4bf890, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/sorter.go:262 +0x2c
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ProcessorBase).moveToTrailingMeta(0xc43921e000)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:693 +0x140
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ProcessorBase).DrainHelper(0xc43921e000, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:640 +0x14f
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*sorterBase).Next(0xc43921e000, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/sorter.go:132 +0x163
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.Run(0x305a6a0, 0xc42b4bf890, 0x30639a0, 0xc43921e000, 0x30377e0, 0xc42f46dc00)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:172 +0x35
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ProcessorBase).Run(0xc43921e000, 0x305a6a0, 0xc4491a14d0, 0xc4420f4888)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:731 +0x98
created by github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).StartAsync
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:587 +0x1e3
```

--- a/pkg/sql/distsql_running.go
+++ b/pkg/sql/distsql_running.go
@@ -709,7 +709,7 @@ func (dsp *DistSQLPlanner) planAndRunSubquery(
 		typ = sqlbase.ColTypeInfoFromColTypes(colTypes)
 	}
 	rows = rowcontainer.NewRowContainer(subqueryMemAccount, typ, 0)
-	defer rows.Close(evalCtx.Ctx())
+	defer rows.Close(ctx)
 
 	subqueryRowReceiver := NewRowResultWriter(rows)
 	subqueryRecv.resultWriter = subqueryRowReceiver

--- a/pkg/sql/distsqlrun/mergejoiner.go
+++ b/pkg/sql/distsqlrun/mergejoiner.go
@@ -226,7 +226,7 @@ func (m *mergeJoiner) nextRow() (sqlbase.EncDatumRow, *ProducerMetadata) {
 		// TODO(paul): Investigate (with benchmarks) whether or not it's
 		// worthwhile to only buffer one row from the right stream per batch
 		// for semi-joins.
-		m.leftRows, m.rightRows, meta = m.streamMerger.NextBatch(m.evalCtx)
+		m.leftRows, m.rightRows, meta = m.streamMerger.NextBatch(m.Ctx, m.evalCtx)
 		if meta != nil {
 			return nil, meta
 		}
@@ -242,7 +242,7 @@ func (m *mergeJoiner) nextRow() (sqlbase.EncDatumRow, *ProducerMetadata) {
 
 func (m *mergeJoiner) close() {
 	if m.InternalClose() {
-		ctx := m.evalCtx.Ctx()
+		ctx := m.Ctx
 		m.streamMerger.close(ctx)
 		m.MemMonitor.Stop(ctx)
 	}

--- a/pkg/sql/distsqlrun/sorter.go
+++ b/pkg/sql/distsqlrun/sorter.go
@@ -142,7 +142,7 @@ func (s *sorterBase) close() {
 		if s.i != nil {
 			s.i.Close()
 		}
-		ctx := s.evalCtx.Ctx()
+		ctx := s.Ctx
 		s.rows.Close(ctx)
 		s.MemMonitor.Stop(ctx)
 		if s.diskMonitor != nil {
@@ -522,7 +522,7 @@ func (s *sortChunksProcessor) chunkCompleted(
 // if a metadata record was encountered). The caller is expected to drain when
 // this returns false.
 func (s *sortChunksProcessor) fill() (bool, error) {
-	ctx := s.evalCtx.Ctx()
+	ctx := s.Ctx
 
 	var meta *ProducerMetadata
 
@@ -592,6 +592,7 @@ func (s *sortChunksProcessor) Start(ctx context.Context) context.Context {
 
 // Next is part of the RowSource interface.
 func (s *sortChunksProcessor) Next() (sqlbase.EncDatumRow, *ProducerMetadata) {
+	ctx := s.Ctx
 	for s.State == StateRunning {
 		ok, err := s.i.Valid()
 		if err != nil {
@@ -600,7 +601,6 @@ func (s *sortChunksProcessor) Next() (sqlbase.EncDatumRow, *ProducerMetadata) {
 		}
 		// If we don't have an active chunk, clear and refill it.
 		if !ok {
-			ctx := s.evalCtx.Ctx()
 			if err := s.rows.UnsafeReset(ctx); err != nil {
 				s.MoveToDraining(err)
 				break

--- a/pkg/sql/distsqlrun/stream_group_accumulator.go
+++ b/pkg/sql/distsqlrun/stream_group_accumulator.go
@@ -65,7 +65,7 @@ func (s *streamGroupAccumulator) start(ctx context.Context) {
 // nextGroup returns the next group from the inputs. The returned slice is not safe
 // to use after the next call to nextGroup.
 func (s *streamGroupAccumulator) nextGroup(
-	evalCtx *tree.EvalContext,
+	ctx context.Context, evalCtx *tree.EvalContext,
 ) ([]sqlbase.EncDatumRow, *ProducerMetadata) {
 	if s.srcConsumed {
 		// If src has been exhausted, then we also must have advanced away from the
@@ -88,7 +88,7 @@ func (s *streamGroupAccumulator) nextGroup(
 			return s.curGroup, nil
 		}
 
-		if err := s.memAcc.Grow(evalCtx.Ctx(), int64(row.Size())); err != nil {
+		if err := s.memAcc.Grow(ctx, int64(row.Size())); err != nil {
 			return nil, &ProducerMetadata{Err: err}
 		}
 		row = s.rowAlloc.CopyRow(row)
@@ -117,7 +117,7 @@ func (s *streamGroupAccumulator) nextGroup(
 			n := len(s.curGroup)
 			ret := s.curGroup[:n:n]
 			s.curGroup = s.curGroup[:0]
-			s.memAcc.Empty(evalCtx.Ctx())
+			s.memAcc.Empty(ctx)
 			s.leftoverRow = row
 			return ret, nil
 		}

--- a/pkg/sql/distsqlrun/stream_merger.go
+++ b/pkg/sql/distsqlrun/stream_merger.go
@@ -49,18 +49,18 @@ func (sm *streamMerger) start(ctx context.Context) {
 // the right stream, all matching on the equality columns. One of the sets can
 // be empty.
 func (sm *streamMerger) NextBatch(
-	evalCtx *tree.EvalContext,
+	ctx context.Context, evalCtx *tree.EvalContext,
 ) ([]sqlbase.EncDatumRow, []sqlbase.EncDatumRow, *ProducerMetadata) {
 	if sm.leftGroup == nil {
 		var meta *ProducerMetadata
-		sm.leftGroup, meta = sm.left.nextGroup(evalCtx)
+		sm.leftGroup, meta = sm.left.nextGroup(ctx, evalCtx)
 		if meta != nil {
 			return nil, nil, meta
 		}
 	}
 	if sm.rightGroup == nil {
 		var meta *ProducerMetadata
-		sm.rightGroup, meta = sm.right.nextGroup(evalCtx)
+		sm.rightGroup, meta = sm.right.nextGroup(ctx, evalCtx)
 		if meta != nil {
 			return nil, nil, meta
 		}

--- a/pkg/sql/executor_statement_metrics.go
+++ b/pkg/sql/executor_statement_metrics.go
@@ -169,7 +169,7 @@ func (ex *connExecutor) recordStatementSummary(
 		sessionAge := phaseTimes[plannerEndExecStmt].
 			Sub(phaseTimes[sessionInit]).Seconds()
 
-		log.Infof(planner.EvalContext().Ctx(),
+		log.Infof(ctx,
 			"query stats: %d rows, %d retries, "+
 				"parse %.2fµs (%.1f%%), "+
 				"plan %.2fµs (%.1f%%), "+

